tensorflow/core/common_runtime/gpu/gpu_init.h

```diff
@@ -24,7 +24,13 @@ class Platform;
 
 namespace tensorflow {
 
-// Initializes the CUDA platform and returns OK if the CUDA
+#if GOOGLE_CUDA
+#define GPU_PLATFORM_NAME "CUDA"
```

Comment:
While I think it unlikely that many folks will have GPUs from two different vendors in the same machine (although it is possible), I think it is likely that folks may wish to build a single *binary* that they deploy on two different kinds of machine. Hence my preference for a compile-time vs run-time test.

It is true that TF programmers would have to avoid hard-coding a particular GPU device in their code, but that's increasingly true anyway — there are a large and growing set of GPU hardware vendors, so my temptation is to say "fix models to allow specifying a target device via, say, a flag".

@zheng-xq do you have an opinion here?

Reply:
@gstoner @dagamayank, could you help weigh in?

@hawkinsp I think shipping one single binary for 2 different system configurations is something we may have to think twice before doing that. Even on PyPI we can see TensorFlow be shipped in multiple packages: `tensorflow`, `tensorflow-aarch64`, `tensorflow-gpu`, etc. It would be natural to deploy TensorFlow on a ROCm-based system with ROCm-based package (ex: `tensorflow-rocm`).

Reply:
Within reason, I'd like to start with "AMDGPU" first, and see how much difference is there. The current "GPU" device is deeply integrated with how NV GPU works. So it much safer to start separately. Once the features are completely, and we see the code is really almost identical, we can consider extend "GPU" to include "AMDGPU" as well.